Source/Scene/ModelExperimental/ClassificationModelDrawCommand.js

```diff
@@ -0,0 +1,339 @@
+import BoundingSphere from "../../Core/BoundingSphere.js";
+import Check from "../../Core/Check.js";
+import defaultValue from "../../Core/defaultValue.js";
+import defined from "../../Core/defined.js";
+import Matrix4 from "../../Core/Matrix4.js";
+import DrawCommand from "../../Renderer/DrawCommand.js";
+import Pass from "../../Renderer/Pass.js";
+import RenderState from "../../Renderer/RenderState.js";
+import BlendingState from "../BlendingState.js";
+import ClassificationType from "../ClassificationType.js";
+import DepthFunction from "../DepthFunction.js";
+import StencilConstants from "../StencilConstants.js";
+import StencilFunction from "../StencilFunction.js";
+import StencilOperation from "../StencilOperation.js";
+
+/**
+ * A wrapper around the draw commands used to render a classification model,
+ * i.e. a {@link ModelExperimental} that classifies another asset. This manages
+ * the derived commands and returns only the necessary commands depending on the
+ * given frame state.
+ *
+ * @param {Object} options An object containing the following options:
+ * @param {DrawCommand} options.command The draw command from which to derive other commands from.
+ * @param {PrimitiveRenderResources} options.primitiveRenderResources The render resources of the primitive associated with the command.
+ *
+ * @alias ClassificationModelDrawCommand
+ * @constructor
+ *
+ * @private
+ */
+function ClassificationModelDrawCommand(options) {
+  options = defaultValue(options, defaultValue.EMPTY_OBJECT);
+
+  const command = options.command;
+  const renderResources = options.primitiveRenderResources;
+
+  //>>includeStart('debug', pragmas.debug);
+  Check.typeOf.object("options.command", command);
+  Check.typeOf.object("options.primitiveRenderResources", renderResources);
+  //>>includeEnd('debug');
+
+  const model = renderResources.model;
+
+  this._command = command;
+  this._model = model;
+  this._runtimePrimitive = renderResources.runtimePrimitive;
+
+  // Classification models aren't supported in 2D mode, so there's no need to
+  // duplicate the model matrix for each derived command.
+  this._modelMatrix = command.modelMatrix;
+  this._boundingVolume = command.boundingVolume;
+  this._cullFace = command.renderState.cull.face;
+
+  const type = model.classificationType;
+  this._classificationType = type;
+
+  // ClassificationType has three values: terrain only, 3D Tiles only, or both.
+  this._classifiesTerrain = type !== ClassificationType.CESIUM_3D_TILE;
+  this._classifies3DTiles = type !== ClassificationType.TERRAIN;
```

Comment:
Minor nitpick: I think the intent would be clearer if we explicitly check for `BOTH` rather than using `!==` here

Reply:
Is it unclear even with the comment? I think it's just going to look messier with the two checks, since the enum names are so long

Reply:
that's fair, I did understand it from the comment. The `!==` just took me by surprise I guess. And it would be more verbose... :thinking: I'd say leave it as-is.